### ISSUE TITLE
msh is broken after ctrl-c (linux)

### ISSUE THREAD

**Issue by debian-user-france1** [gekigek99/minecraft-server-hibernation#113]
When I stop msh using ctrl-c it successfully shuts down mc server and says: exiting msh
But then msh doesn't create a listener at start. I have to reboot (killall bash also worked) and rebuild msh else it will not work anymore!
I don't know if I really have to reboot or killall bash but I did that:

1. time this bug: msh broken, reboot, still broken, rm -r msh && go build . - works
2.  time this bug: msh broken, some rebuilds and a re-download into another folder (I copied config.json), still broken, killall bash, a bit waiting time, rebuild - works

I prob. don't know why this happens.
The 2. time I had to press ctrl-c because I just typed mine without anything wich caused the script to not execute anything written = can't stop server (ANOTHER BUG)

**Comment by debian-user-france1** [comment]
[that might be it](https://github.com/gekigek99/minecraft-server-hibernation/issues/104)

**Comment by debian-user-france1** [comment]
Ok the problem above is likely the same.
It fails to check for updates. Please - if http-request fails or like so then print out: Cheking for updates failed and skip this part.

**Comment by debian-user-france1** [comment]
#104

**Comment by gekigek99** [comment]
What version are you using?

> The 2. time I had to press ctrl-c because I just typed mine without anything wich caused the script to not execute anything written = can't stop server (ANOTHER BUG)

with server online if i type `mine` i get this (`msh command error: specify mine command`) which is to be expected:  
(i have debug set to false)
```
[23:41:51] [Server thread/INFO]: Done (13.860s)! For help, type "help"
2021/07/28 23:41:51 *** MINECRAFT SERVER IS ONLINE!
mine
msh command error: specify mine command
```

**Comment by gekigek99** [comment]
> time this bug: msh broken, reboot, still broken, rm -r msh && go build . - works
    time this bug: msh broken, some rebuilds and a re-download into another folder (I copied config.json), still broken, killall bash, a bit waiting time, rebuild - works

this sounds like msh is unable to launch the minecraft server because an istance is already running (this is an example log):
```
2021/07/28 23:45:00 *** listening for new clients to connect on 0.0.0.0:25555 ...
msh start
2021/07/28 23:45:08 *** MINECRAFT SERVER IS STARTING!
2021-07-28 23:45:08,788 main ERROR Unable to delete file C:\Users\Giacomo\Desktop\minecraft server\logs\latest.log: java.nio.file.FileSystemException C:\Users\Giacomo\Desktop\minecraft server\logs\latest.log: Impossibile accedere al file. Il file � utilizzato da un altro processo
[23:45:18] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[23:45:18] [main/FATAL]: Failed to start the minecraft server
java.io.IOException: Impossibile accedere al file. Un altro processo ha bloccato una parte del file
        at sun.nio.ch.FileDispatcherImpl.write0(Native Method) ~[?:?]
        at sun.nio.ch.FileDispatcherImpl.write(FileDispatcherImpl.java:68) ~[?:?]
        at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:113) ~[?:?]
        at sun.nio.ch.IOUtil.write(IOUtil.java:58) ~[?:?]
        at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:285) ~[?:?]
        at agn.a(SourceFile:40) ~[server.jar:?]
        at dib$a.<init>(SourceFile:296) ~[server.jar:?]
        at dib.c(SourceFile:284) ~[server.jar:?]
        at net.minecraft.server.Main.main(SourceFile:123) [server.jar:?]
2021/07/28 23:45:18 *** MINECRAFT SERVER IS OFFLINE!
```

Is it the case?

**Comment by gekigek99** [comment]
> It fails to check for updates. Please - if http-request fails or like so then print out: Cheking for updates failed and skip this part.

I'll correct this

**Comment by gekigek99** [comment]
> #104

this similar bug you are experiencing happens always or only sometimes?

(in the next weeks the host will take down the server because of problems so it might be due to that... it's still strange that this little problem causes msh to do such strange things... after 4 seconds the request should timeot and everyone should continue to live happily)

**Comment by debian-user-france1** [comment]
> What version are you using?
> 
> > The 2. time I had to press ctrl-c because I just typed mine without anything wich caused the script to not execute anything written = can't stop server (ANOTHER BUG)
> 
> with server online if i type `mine` i get this (`msh command error: specify mine command`) which is to be expected:
> (i have debug set to false)
> 
> ```
> [23:41:51] [Server thread/INFO]: Done (13.860s)! For help, type "help"
> 2021/07/28 23:41:51 *** MINECRAFT SERVER IS ONLINE!
> mine
> msh command error: specify mine command
> ```

well that didn't  happen to me. Maybe Linux bugged and wrote a special thing so it thought there comes more. Like ' but unreadable by the shell. Newly happens more often and can f* a whole script

**Comment by debian-user-france1** [comment]
> > time this bug: msh broken, reboot, still broken, rm -r msh && go build . - works
> > time this bug: msh broken, some rebuilds and a re-download into another folder (I copied config.json), still broken, killall bash, a bit waiting time, rebuild - works
> 
> this sounds like msh is unable to launch the minecraft server because an instance is already running (this is an example log):
> 
> ```
> 2021/07/28 23:45:00 *** listening for new clients to connect on 0.0.0.0:25555 ...
> msh start
> 2021/07/28 23:45:08 *** MINECRAFT SERVER IS STARTING!
> 2021-07-28 23:45:08,788 main ERROR Unable to delete file C:\Users\gekigek99\Desktop\minecraft server\logs\latest.log: java.nio.file.FileSystemException C:\Users\gekigek99\Desktop\minecraft server\logs\latest.log: Impossibile accedere al file. Il file � utilizzato da un altro processo
> [23:45:18] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
> [23:45:18] [main/FATAL]: Failed to start the minecraft server
> java.io.IOException: Impossibile accedere al file. Un altro processo ha bloccato una parte del file
>         at sun.nio.ch.FileDispatcherImpl.write0(Native Method) ~[?:?]
>         at sun.nio.ch.FileDispatcherImpl.write(FileDispatcherImpl.java:68) ~[?:?]
>         at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:113) ~[?:?]
>         at sun.nio.ch.IOUtil.write(IOUtil.java:58) ~[?:?]
>         at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:285) ~[?:?]
>         at agn.a(SourceFile:40) ~[server.jar:?]
>         at dib$a.<init>(SourceFile:296) ~[server.jar:?]
>         at dib.c(SourceFile:284) ~[server.jar:?]
>         at net.minecraft.server.Main.main(SourceFile:123) [server.jar:?]
> 2021/07/28 23:45:18 *** MINECRAFT SERVER IS OFFLINE!
> ```
> 
> Is it the case?

No wrong. Minecraft Server is starting, then the plugin Geyser started and loaded up. This shows an output similar to the Minecraft output when server started. So your script thinks that the server is started. When now someone joins he gets kicked but your script does not recognize that. It counts one up. Sometimes I have like 3 more users that normal because people just keep joining. Also this causes then to not freeze the server anymore except I write msh freeze or mine reload

**Comment by debian-user-france1** [comment]
Wait are we now in 'msh thinks mc server started ...' or msh is broken after ...

**Comment by debian-user-france1** [comment]
> > #104
> 
> this similar bug you are experiencing happens always or only sometimes?
> 
> (in the next weeks the host will take down the server because of problems so it might be due to that... it's still strange that this little problem causes msh to do such strange things... after 4 seconds the request should timeot and everyone should continue to live happily)

Well it looks like this is not working prob. 
Because there the http-request failed in this bug and script just crashed. For me if I quit msh (newly just restart it) it can happen that it crashes. Could you check your script and see if there might be anything that it doen't continue when http request fails (not timeout you can see that in this bug report)

**Comment by debian-user-france1** [comment]
could you please add the 'bug' label? thks

**Comment by gekigek99** [comment]
> Wait are we now in 'msh thinks mc server started ...' or msh is broken after ...

let's keep things organized please, here we can speak about `msh is broken after ctrl-c (linux)`

**Comment by gekigek99** [comment]
> Because there the http-request failed in this bug and script just crashed. For me if I quit msh (newly just restart it) it can happen that it crashes. Could you check your script and see if there might be anything that it doen't continue when http request fails (not timeout you can see that in this bug report)

before assuming it's caused by the http request (i have no other reports of that... so it might as well be a problem in your server config... what hosting service are you using by the way?)

I would start by checking if the http request is causing the issue:

1) download the dev branch [here](https://github.com/gekigek99/minecraft-server-hibernation/tree/dev)

2) then replace these lines

https://github.com/gekigek99/minecraft-server-hibernation/blob/13d23da12a7834c94497611be53b98002878a913/minecraft-server-hibernation.go#L53-L56

with these (comment out these 4 lines):

```go
// // launch update manager to check for updates
// go progctrl.UpdateManager(version)
// // wait for the initial update check
// <-progctrl.CheckedUpdateC
```

3) then run `go build .`
4) set the debug option to `true`
5) check if the problem persist

_please report the full log to speed up things thanks_

**Comment by debian-user-france1** [comment]
https://cloud.bubicloud.ch/nextcloud/index.php/s/LBMJLA6AL2pdLk2 <- log (It just worked but error)

2. why do I have to edit this? Couldn't I just set checkforupdates to false?

**Comment by gekigek99** [comment]
> log (It just worked but error)

what do you mean? where can I see the error? (I don't see any ctrl-c related error)

> why do I have to edit this? Couldn't I just set checkforupdates to false?

in the dev branch some modifications where done to that part of code and it is easier to debug how I described it to you

**Comment by debian-user-france1** [comment]
Could you also change the command that msh executes every time when a player joins (I think it is then)?
The command that msh uses: /list (to list online players)
The command that is correct: list (without dash)

In 1.16/17 (I guess in every mc version) every command that is executed can't be correctly detected when it has a /

**Comment by gekigek99** [comment]
It was already solved in commit 13d23da12a7834c94497611be53b98002878a913

Use the dev branch to access the most recent updates

Can I close this issue now?  
You can open a new issue for bugs/errors or a new discussion for general problems.
